### DISCUSSION TITLE
R017 Prerelease

### DISCUSSION THREAD

**Discussion by eauerbach** [CMRR-C2P/MB#288] [Announcement]
This thread can be used for discussions of the R017 prerelease versions which are now available on the [main CMRR MB C2P web site](https://www.cmrr.umn.edu/multiband/).

**R017pre15** is the latest prerelease version. It is currently available for **VE11C**, **VE11E**, **VE12U-SP01**, **VE12U-AP01**, **VE12U-AP02**, **VE12U-AP04**, **XA20A**, **XA30A**, **XA31A**, **XA50A**, **XA51A**, **XA60A**, **XA61A**, **XA61A-SP01**, **XA70A**, and **XB10A**.

Please refer to the [Release Notes](https://downloads.cmrr.umn.edu/software/file2/4478) for detailed information about the changes between versions.

Known issues:
- For **XA30** and **XA31** versions: **Do not attempt to install the **CMRR_C2P_eja.cer** security certificate unless you are certain that security patch SD05 (or higher) is installed**. Please consult the **Install_Instructions_XA.pdf** document for details about how to identify the patch level. Installing certificates on a system with a lower patch level can cause the scanner to become unbootable.
- For all **Numaris X** versions: Note that it is only necessary to install the **CMRR_C2P_eja.cer** security certificate once per scanner. Certificate installation is permanent; it is not necessary to reinstall the certificate when upgrading between sequence versions or when installing other C2P packages from CMRR that use the same certificate (e.g., Spectroscopy Tools).
- For **XA30**: If using the "High dynamic range" DICOM format (16-bit) option on **XA30**, it is critical to make sure the **SP02 patch** is installed and to manually close all View&GO tabs/windows and ensure they remain closed throughout acquisition. Otherwise the system will likely crash. There does not seem to be a permanent solution to this other than a future patch from Siemens. It is expected that this issue is fixed in XA31 and newer versions.

**Comment by jpyles** [comment]
@eauerbach For those of us sticking with VE11C for the moment -- is there any reason to upgrade to R017? And will the license for R016a be extended beyond 30 January, 2023 if we are still on VE11C? 

Thanks! 

--John

**Reply by rwmair** [reply]
John - download the R017 instructions from the CMRR website and take a look at the change-log. There's a whole host of new features in R017 aside from the transition to XA platforms. Some may be useful for some of your users. But, for preexisting protocols/studies running on VE11C, there's no reason to upgrade at this time, and I don't know of a VE11C site planning to move up to R017 at this point. Because of the time spent on the issues with XA30, Eddie was unable to test the VE11C R017 release fully - hence the "use at own risk" warning. 
He's on vacation, so it might be a little while before he responds on the license issue. I'd expect the support for R016a would continue, but if they get to test R017 fully for all platforms in the fall, maybe they'll force the transition to R017 as used to be the case for prior versions. Tho hopefully there's the option to retain R016a until our scanners inevitably move to XA30 to reduce the number of sequence version changes our users experience.

Ross

**Comment by TimmWetzel** [comment]
In https://github.com/CMRR-C2P/MB/issues/296#issuecomment-1219846406 @eauerbach mentioned that the XA30A-SP02 service pack might fix the XA30 instability issues with HDR and physiological DICOM data.

Can anyone confirm this, especially for long acquisitions (>2000 repetitions)?
(I am also interested in any other experiences with XA30-SP02, whether good or bad — the installation of XA30-SP02 is scheduled for our next maintenance slot, and I have no information yet about what changes to expect.)

Regards,
Timm

**Reply by bbfrederick** [reply]
We got XA30A-SP02 installed this week - from my initial testing it does not seem to fix the DICOM physio recording (nothing came up on the inline display, recon never completed, and I had to reboot the scanner). I tried it both with the PPO only, and the PPO and EKG modules in the room (I saw a discussion online saying that if you don’t record all channels in the continuous mode, the physio files are unreadable).  I'm hoping to check the HDR a little later today.

XA30A-SP02 does at least raise the flB1Limit on the 64 channel head coil from 24uT to 30uT (but not back to 32uT, which is where it is on VE11), which helps a lot with pulse clipping.

Blaise

**Reply by TimmWetzel** [reply]
In the light of the rather quiet response I am assuming that the XA30A-SP02 service pack is unfortunately not a game changer regarding HDR and physiological DICOM data.

But I would love to be proven wrong …

Regards,
Timm

**Reply by eauerbach** [reply]
Our experience lately has been that since SP02 if you consistently keep the View&Go window closed, the scanner does not crash. That is, make sure the auto load option is disabled in every protocol, and make sure to close any and all existing View&Go windows before scanning.

**Reply by mharms** [reply]
So, @eauerbach, to try to draw this out a bit, with SP02, use of 16 bit DICOM is fine, but only if the View&Go is closed?  And the DICOM physio recording?

**Reply by eauerbach** [reply]
As far as I know, both 16-bit and DICOM physio are fine to use on XA30A-SP02 as long as View&Go is always closed.

<sub>(I still wouldn't recommend for anyone with a functional VE11C or VE11E scanner to upgrade to XA30—wait for XA60 or so next year if you can.)</sub>

**Reply by rwmair** [reply]
Thanks @eauerbach! And to draw it out a bit more, what's the status for XA31 and XA50 (if tested) with these issues? As this will hopefully extrapolate better to XA60. Are 16-bit and DICOM physio on by default for the versions after XA30? And do they also require View and Go to always be closed; or is normal functionality restored with the later XA software versions?

**Reply by eauerbach** [reply]
16-bit and DICOM physio are only disabled by default right now for XA30, but I will eliminate that with the next release. I haven't received any feedback about XA31, and XA50 isn't quite ready yet.

**Reply by rwmair** [reply]
Thanks! Hopefully someone who is running R017pre5 on a XA31 system can provide some feedback on their experiences.

**Comment by rwmair** [comment]
Reviving the question from @jpyles on July 15 regarding R017 for VE11C. Has anybody with a VE11C system upgraded from R016a to R017? John, did you go this route? @eauerbach - have you had any users download and provide feedback on the VE11C R017 release?

I have one research group at one site that wants to take advantage of the much-increased max number of BOLD timepoints feature. Everyone else is nervous about making a change they otherwise don't need to at this point - which I understand!

**Reply by jpyles** [reply]
@rwmair We did not upgrade. We're planning to stick with R016a until we upgrade to VA software. So long as @eauerbach extends the license past January 2023. :)

**Reply by rwmair** [reply]
Thanks @jpyles. I figured that's probably what you did. And, yes, eagerly awaiting news of the license extension beyond Jan 2023.

**Reply by bpinsard** [reply]
At UNF Montreal, we upgraded to the R017 on Prisma Fit VE11C ~ 3 months ago, and it went very smoothly. We decided to try it for the extended number of BOLD volumes for a sleep study, and knowing that it would be possible to rollback to the previous version.
For the Neuromod project, we scan ~10 times a week with ~5 BOLD runs in each session, also acquired some multi-echo pilots, other labs used it, and had no problems so far. We also scan DWI a bit less regularly, no problems were reported.

**Reply by rwmair** [reply]
@bpinsard - Thanks so much for this news confirming R017 worked well without any issues on your VE11C system.

**Reply by rwmair** [reply]
@bpinsard - Just wondering if things are still working well with R017 on your VE11C system, without any issues?  I saw this issue posted: https://github.com/CMRR-C2P/MB/issues/308 and wondered if it was your system, or a different VE11C scanner with R017? Thanks!

**Reply by bpinsard** [reply]
Yes it is our system, just happened once months after upgrading and we use the CMRR sequence a lot.

**Reply by rwmair** [reply]
@bpinsard - Thanks again for the confirmation. Good to hear R017 on VE11C is working well - tho if you have any reconstruction issues in the future, it would be great if you could report it here.  Did you do any comparison tests between R016a and R017 around the time of the change-over?  Thanks

**Reply by bpinsard** [reply]
>  tho if you have any reconstruction issues in the future, it would be great if you could report it here.

Sure will do.
 
> Did you do any comparison tests between R016a and R017 around the time of the change-over? Thanks

No, that was a pretty wild move. But we have the same participants scanned 1h ~bi-weekly for years before and months after, same parameters, approximate same head(CaseForge)/FoV positioning, doing similar tasks, so if there are any analysis/QC/checks that you want me to run, let me know.

**Reply by whitecraine** [reply]
Hi @eauerbach! In part to celebrate the new year of 2024, I tried installing R017 (4/25/2023 build) on a VE11C SP04 3T Skyra. Unfortunately, it is causing multiband enabled protocols to fail with reconstruction error. I tried restartmr as well as installing after a full shutdown but result is the same. I also tried importing the seq from default and confirmed it can recon with GRAPPA but not with MB enabled. Each time system was in Update Mode and the install dialog box claims to be successful for all seq and ice files. In viewing the logviewer, some potential error messages include: 

getMaskFromRegistry ERROR: Could not open reg key "DEBUG_SEQ_B1CL"
ERROR: It has been tried to set an nucleus which is not allowed for this measurement 
FilterGraph::SetProperty -- functor ejaMBImageShuffler does not support property ICEWin32SVNRevisioon
FilterGraph::SetProperty -- functor ejaMBImageShuffler does not support property SequenceSVNRevision
FilterGraph::SetProperty -- functor ehaMBScanDemultiplexer does not support property ApplySBRefFlagsFix
ERROR: Failed to create an EQEvent object for GradPulseTrap 'EPI RO Blip3D' (type GRAD_PULSE_TRAP) for GPA

Reinstalling R016 went smoothly and was the only way to get the MB protocols working again. Any suggestions would be greatly appreciated!

**Reply by bpinsard** [reply]
This reminds me that we had a reconstruction error with R017pre6 (4/25/2023 build) on PrismaFit VE11C for DWI and fMRI that required us to rollback to R017pre5. We never attempted again the upgrade to R017pre6 so far. 
I don't know if you can still access the pre5 version (forging URL from the pre6 one) but that might be worth trying.

**Reply by whitecraine** [reply]
Unfortunately, clicking the "download now" button for pre5 doesn't do anything :( Any chance you still have the pre5.zip file you could share?

**Reply by eauerbach** [reply]
That error 100% means the old R016 ICE dll is still in use somehow.

If you are absolutely sure all of the ICE files in C:\Medcom\MriCustomer\Ice are getting updated correctly, and rebooting the host does not change anything, then another possibility is if there are multiple conflicting copies of the binaries on the system, e.g. if at some point someone manually copied binaries to the old system paths.

Please make sure there are no CMRR_MB* files in C:\Medcom\bin and there are no libCMRR_MB* files in C:\Medcom\MCIR\Med\lib. Those paths should no longer be used for customer binaries, so the installer will not touch anything there. But since they are the system paths they are higher priority than the customer paths and will override anything in the proper location.

Unfortunately the installer does not check for this situation; I should probably update that or make a clean uninstall tool that does a search and destroy of every possible trace...

**Reply by whitecraine** [reply]
Hi @eauerbach, 

I confirmed that there were no old C2P files in the C:\Medcom\bin or C:\Medcom\MCIR\Med\lib. What would be the next step? Should I try deleting all the CMRR_MB and libCMRR files in the C:\Medcom\MriCustomer\Ice location prior to installing R017? I tried a fresh install of R017 and confirmed the new ice files were there but the sequence still didn't work when MB was turned on (even after full shutdown and reboot). In case it helps, I uploaded the savelog files to:

https://ucsf.box.com/s/cs03nxocj5k394e6yxixzi7sgqri68ib

**Reply by whitecraine** [reply]
Hi @eauerbach, if it would be helpful, could. you send me a copy of the R017pre5 version to help with troubleshooting? That seems to have worked for another VE11C site.

**Reply by whitecraine** [reply]
Hi @eauerbach I tried the install process once more after the Siemens preventative maintenance but still facing the same errors. Would it be possible for me to try the R017pre5 version given @bpinsard has shown this to be an effective interim solution on his PrismaFit VE11C? I tried downloading it from this modified link https://www.cmrr.umn.edu/multiband/Downloads-Seq/CMRR_MB_EPI_VE11C_R017pre5.zip, however after clicking the download pre5.zip button, it just brings me back to the main cmrr c2p webpage.

**Reply by eauerbach** [reply]
No, there is no going back to pre5, there can only be an old copy of a R016 ICE dll somewhere that is not getting updated.

I have finally finished writing a comprehensive uninstall/clean tool; can you please try running that to see if it turns up any issues. It should only be necessary to run it once, but if you have some persistent issues you can run it as many times as you like. In the end it should report that it finds nothing. Once all of the old files are cleaned up you should be able to install any version without issue.

https://www.cmrr.umn.edu/multiband/CLEAN_UNINSTALL_CMRR_MB_EPI.exe

**Reply by whitecraine** [reply]
Hi Eddie, 

I was able to successfully run the uninstaller after system reboot. Unfortunately, the recon still fails anytime MB > 1. Let me know if I should upload the latest SaveLog or if there is anything else I should try. 

Joseph

![IMG_3137 2](https://github.com/CMRR-C2P/MB/assets/2513968/beea60e6-5e16-444f-b90c-c8a82d616933)
![IMG_3140](https://github.com/CMRR-C2P/MB/assets/2513968/3c0be8f7-8ded-4d8b-9b4f-68ac3ffee6bf)
![IMG_3141](https://github.com/CMRR-C2P/MB/assets/2513968/547d33a9-03c9-48ba-8b86-d3c6a2422d73)

**Reply by eauerbach** [reply]
I still can't understand why the CMRR_MBIcePrg2D.dll is not getting updated. It says the delete succeeded, then presumably you ran the new installer and after that saw the error again? But the error is definitely still coming from the old dll.

Could you try running the uninstaller, then rebooting the host, then running the uninstaller again? If CMRR_MBIcePrg2D.dll is still there the second time then something is wrong. I wonder if something strange is happening which is restoring the old dll in the background.

Another possibility would be to rename it. So: run the uninstaller, reboot the host, run the uninstaller again. If CMRR_MBIcePrg2D.dll is still there, or if you still see the error, rename CMRR_MBIcePrg2D.dll to CMRR_MBIcePrg2D.dll.old or something like that. Then reboot the host again. If CMRR_MBIcePrg2D.dll returns again after that reboot, an exorcist may be required.

**Reply by whitecraine** [reply]
I was finally able to install pre9 (after a full shutdown and switching on Embedded Control Mode prior to running the clean uninstaller). Although the first MB EPI scan failed due to a recon error after making the SBref image, subsequent re-scans worked fine! Thanks so much!

**Comment by xiangruili** [comment]
We are planning to upgrade the PrismaFit to XA30A+R17, and wonder if there is any other consideration to delay it.
Apart from the physio recording issue mentioned , is there any other issue with XA30A+R17?
Thanks.

**Comment by eauerbach** [comment]
I am uploading the new R017pre6 releases now.
- Support is continued for all of the previous versions, and newly added for VE11E, VE12U-AP02, and XA50.
- The restrictions on 16-bit DICOM and DICOM physiology on XA30 are removed, but be sure you have the SP02 patch installed and keep the the View&GO windows closed while scanning for the best chance at system stability.
- A new option "delay before PC" has been added on the Sequence/Special card to insert an optional delay between prephasing gradients and the first phase correction readout. This was requested for head gradient systems; for everyone else it should be left at the default of 0.
- For Magn/Phas reconstructions, phase images are now always split into separate series, even for single-band reference scans (this will avoid the Mixed image type issue on XA platforms). The phase series are labeled with a "_Pha" suffix.
- An adjustment has been made to the saving of DICOM physiology images which hopefully should improve the reconstruction performance and stability on XA30/31/50.

**Comment by eauerbach** [comment]
I am uploading the new R017pre7 releases now (only for XA versions at the moment).

- The installer has been updated. It is a bit prettier (more GUI elements), and it should be more reliable at detecting the scanner version (previously a problem on XA50) and the IDEA license status.
- Various issues around protocol conversion have been fixed in the BOLD sequence. This should also address the crash seen when the stimulation monitor attempts to adjust gradient rise times due to limits exceeded.
- The long reconstruction delay when saving DICOM physiology images on XA versions should be fixed.
- It should be possible to save DICOM physiology on XA31 and XA50 now without an error stop at the start of the scan.
- Automatic frequency adjustment at the start of measurement has been restored on XA.
- Updated install documentation to describe how to check for the SD05 patch from Siemens, which fixes the catastrophic issue of the scanner becoming unbootable when installing security certificates.
- "Legacy" physiology logging should now work as expected on XA (although it is still not recommended).

**Reply by eauerbach** [reply]
Also a housekeeping note; this thread was getting pretty long, so to keep it manageable I deleted a lot of old messages. I converted most of the messages that described still open issues into...issues, and deleted messages referring to old versions or solved issues. I am not sure if everyone gets notifications when this happens. In any case, please feel free to make new comments and open new issues, I will prune periodically.

**Reply by TimmWetzel** [reply]
Hello @eauerbach,
we are considering to update from pre5 to pre7 on a XA30A Prisma. 
We have some longitudinal studies running — can we assume that image characteristics for existing protocols will be completely unaffected? 

According to the release statements above, most changes should be performance-related or optional, but for example the frequency adjustment fix might be an exception.

**Reply by eauerbach** [reply]
Yes, there is no reason not to update

**Reply by sjayejohnston** [reply]
Hi

We would like to try the software (SIemens running XA60). Would it be possible to be allowed access? We have the C2P in place for the spectroscopy packages but the website does not want me to download this package. 

Sorry if this is the wrong place, but I read that to gain access I need to contribute to this thread.

S

**Reply by rwmair** [reply]
Hi,
I think the C2P request process for the CMRR multiband package still starts here:
https://license.umn.edu/product/cmrr---center-for-magnetic-resonance-research-software-for-siemens-mri-scanners

Or possibly its on the Siemens C2P Exchange board these days.

Ross

**Comment by abfeld007** [comment]
![IMG_3325](https://github.com/CMRR-C2P/MB/assets/36004922/e8136391-0499-4021-a825-b9ede9f41877)

With SD05 and the new installer released for XA30 we were able to successfully install the multi-band sequence and go it running. But we had to rename two files to do so. I'm not sure if it is an error in the naming of the file, or it should have truly had different library files. It was looking for libcuda.so.1, but we had lubcuda.so.418.67 so we renamed it to .1. We also renamed the libnvidia file.

**Reply by eauerbach** [reply]
Did the MaRS reboot as part of the install? You should not have to rename anything, some links should be created on the MaRS automatically when it reboots.

**Reply by abfeld007** [reply]
The command line prompt asking if we would like to reboot MaRS did come up we said yes. It rebooted the MaRS, but this error occurred afterwards.

**Reply by eauerbach** [reply]
Was this a new install on this system, or were you upgrading from a previous version (e.g. R017pre6)? And do you know if you actually have a GPU in the MaRS? Prisma systems should all have a GPU, but upgraded Skyra systems might not.

**Reply by abfeld007** [reply]
This was a new install, as it was deleted when our scanner was upgraded from VE11c to XA30. We are using a Skyra. I am not sure if it has a GPU

**Reply by eauerbach** [reply]
Ah! That makes sense. In previous versions I had assumed all XA systems received a MaRS upgrade including a GPU, but since I have learned that Skyra upgrades do not. So I have to install a dummy GPU driver to make my ICE program happy. I thought the latest installer should have addressed this automatically, but apparently my fix is not working. So thanks for pointing that out, I will work on it. In the meantime if renaming the files works, that is fine.

**Reply by eauerbach** [reply]
reference issue #325

**Reply by abfeld007** [reply]
Thank you! I'm glad its finally back on our scanner.

**Comment by xiangruili** [comment]
After postponed several times, we finally upgraded to XA30. The R017 Prerelease installation was smooth. I have two questions.

First, the sequences for ongoing studies were converted successfully. Should we use the converted sequences, or build new ones from those supplied by the installation package? Or there is no much difference?

Second question is unrelated to MB but to XA30 software or Windows 10. Hope people don't mind I ask here. We have been using DICOM Online Export for a long time, like what is needed for FIRMM. After XA30 upgrade, the system won't send images to the shared network drive, while we can read/write the drive from both command line and File Explorer. It seems the mechanism to set up the export via ideacmdtool and IceConfig.evp are the same as before. Siemens gave us the instruction to permanently map a network drive to the Windows 10 system, and, with some network policy exception, we made it work, but as our tech predicted, it did not solve the issue. Has anyone experienced similar issue after the upgrade? 

Thanks and happy new year!

**Reply by mharms** [reply]
Regarding your first question, I would suggest reviewing the "Upgrade Info" (I think that's what it is called) associated with each scan after conversion, and see if there were any changes made as part of the upgrade/conversion, and if so, whether you are fine with them.

**Reply by xiangruili** [reply]
Thanks @mharms for the information. The only changed setting is Coil Selection, so it should not be a concern.

To my second question: the one-step setup for network share seems dead for the XA30 version. The BOLD Add-in can be used to transfer images in real time, but it will need to be set up for each sequence. Also the user name should be in format of domain\user, otherwise it will likely fail.

**Comment by nie-xingju** [comment]
Thanks and recently we upgraded to R017pre6 in our Siemens 3T PRISMA ( VE11E). 
here is my question about fat saturation. it says in release that " Added UI options to manually configure fat saturation frequency offset and nominal flip angle". However I could not locate this function in any tab? Could you please help with this?

**Comment by Lestropie** [comment]
I have a few different pieces of feedback after having installed R017pre9 on our Siemens Vida XA50.

1.  Image reconstruction fails for SE sequence with SENSE1 on and magnitude + phase reconstruction requested. No such problem on DIFF sequence. Disabling phase image export resolves. Possibly, but not yet confirmed myself (based on external report), reverting to SoS rather than SENSE1 resolves.

2.  Considerable console system lag, in two key respects:
    1.  Any protocol editing card that shows TE & TR is slow to respond, both on first display and particularly if attempting to edit one of those two fields, maybe on the order of 5-10 seconds. I'm guessing that the computational cost of the advanced gradient performance model is the reason why the Siemens product sequence restricts the search domains of these parameters to 100ms and 1ms increments respectively, rather than CMRR's 1ms and 0.2ms.

        (Personally, for TR specifically, I set the acquisition time per slice stack to a multiple of 5ms, multiply by number of slice stacks, then add the 20ms frequency adjust per volume. This mitigates imprecision in DICOM slice timing reporting. If you have to reduce the sequence's TR search space to reduce system lag, this might be a better way to do it, rather than round 100ms increments like the product sequence?)

    3.  With one CMRR sequence executing, and attempting to copy the centre of slice stack from one CMRR sequence to another, the console froze for multiple minutes, on multiple occasions. The sequence continued to execute, but the console was unresponsive, sometimes not even updating the indicated remaining acquisition time for the currently executing sequence. Have not yet had the chance to try to replicate to diagnose the exact circumstances.

4.  "Dark voxel" issue reported in #278 not fully resolved. I unfortunately don't have direct comparison against the old sequence version on this particular magnet. Phase is definitely smooth across slices, [as expected](https://github.com/CMRR-C2P/MB/issues/278#issuecomment-2118148292), but that doesn't seem to prevent entirely these singularities.
    **Edit**: Moved exemplar images to [comment](https://github.com/CMRR-C2P/MB/issues/278#issuecomment-2246749194) in #278.

**Reply by eauerbach** [reply]
1. I don't think this has to do with the sequence variant, I have seen it in the diffusion sequence. It is a bug in the new SENSE1 code when phase images are enabled, and seems to be a timing issue related to the number of slices vs. the number of reconstruction threads. Anyway, I am aware of this issue and should have it fixed in the next version.
2. Unfortunately these sound like "normal" stability issues on the XA platform, seems to be better with the newer service packs and newer versions like XA60, the same code is not problematic on VE...
3. I feel like things are improved but not entirely fixed, so it would be nice to save the raw data when you run into a particularly obvious example.

**Comment by whitecraine** [comment]
After upgrading to R017 on my VE11C Skyra, protocols using grappa with GRE ref scan no longer work (all parameters are grayed out). Trying to rebuild the protocol from scratch goes fine until trying to select GRE ref in the end (see attached screenshot) so I switched to FLEET. 

![IMG_4454](https://github.com/user-attachments/assets/a1447fa5-0524-4bd8-ad86-929c28404546)

**Comment by jberrebi** [comment]
Hi,
I have downloaded the pre9 version for XA60 and installed the new binaries on our Prisma 3T for getting physiology data. That works well. Thank you!

However, I get a warning from the stimulation monitor on sequences with parameters that run without warning on Siemens sequence.
What we run on the R2017 pre9 are the standard parameters recommended by Siemens, which means:
GRAPPA 2
multiband 2
2mm resolution isotropic (or 2,2mm)
100 points on FOV 200mm (or similar)
The stimulation monitor systematically pops up and I make the choice of changing the gradients. But can it be done automatically? And is it normal to get warnings for too high stimulation on CMRR while there is no warning on Siemens' sequence?

When I choose 2,5mm resolution, I don't get the warning anymore.

Thank you,
Jonathan

**Reply by jberrebi** [reply]
The issue is systematic only on fMRI, not on DTI. For fMRI I just use gradient mode "fast" instead of "performance" and we stop getting the pop-up which forced us to take gradient mode "fast". As long as I can keep the values of parameters like TR, resolution, etc after switching from "performance" to "fast", are there any downsides of changing the default value for gradient mode?

**Reply by rwmair** [reply]
I would say no downsides, but the benefit of not having stimulation warnings. For fMRI on a Prisma, using Performance Mode only impacts the slew rate of the gradient pulses in the EPI readout-train. The actual max g value is pegged in the software, only changing for the diffusion-weighting pulses in diffusion sequences. When we first upgraded our Trio to Prisma, I found it was very easy to induce a stimulation warning using performance mode. Or even just induce PNS in subjects without a stimulation warning before the scan ran. Since then, I've tended to reserve Performance Mode for fMRI scans where spatial resolution of 2.0mm or less is needed - in which case the matrix size is large enough that the echo spacing remains fairly high. But on coarser resolutions, PNS tended to result. Although others do use Performance for 2.2-2.4mm resolution. 

If you're trying to match a given protocol, and that protocol uses Performance mode, you'll need to use Performance as well to match the echo-spacing, which impacts signal distortion. If you don't care about that, you'll probably still need Performance unless the original protocol leaves a lot of leeway in timing parameters. If TE and TR are minimized, then you'll need the reduced echo spacing from using Performance mode to get there.

One other option to get there is set the protocol to use Performance* mode. You can't choose this in the normal course of editing, but if you run a protocol and the PNS warning pop-up appears and gives in option to "reduce slew rate" as an option instead of making a crazy change to FOV, you can choose the "reduce slew rate" option. This will keep Performance mode but reduce the slew rates some what. Its not necessarily stepping back to Fast mode tho. Once the scan has run, you can drag it into DotCockpit and save the extracted protocol that will show Performance* as the gradient mode.  This may still result in a change in echo spacing by 0.01 or 0.02 ms however.

**Reply by jberrebi** [reply]
Thanks a lot for you answer rwmair!

**Comment by jberrebi** [comment]
I have two questions regarding the box "suppress 16-bit DICOM".

1.   Are there any downsides in checking that box? I doubt it changes the number of bits during sampling by the the A/D converter, but just to check: is this the case? I've checked the DICOM header and even when checking that box, 16 bits are allocated, but 12 are used for storage (like in Siemens vendor bold sequence).
2. The warning in the CMRR's documentation about FFT-scale factor is just for those who would like to have a factor larger than 1 for using the dynamic range more efficiently, isn't it? Or should we even check for clipping with FFT scale factor equal to 1?

**Reply by jberrebi** [reply]
According to Siemens datasheet, the receiver's signal resolution is 32-bit, so I guess that we actually lose bitwise resolution by checking the "supress 16-bit DICOM" box.

**Reply by mharms** [reply]
Not sure I follow.  If you select "Suppress 16-bit DICOM", you'll get 12-bit DICOM, which have considerably less dynamic range than 16-bit DICOMs.  How is that losing resolution?

**Reply by jberrebi** [reply]
I wrote bitwise resolution, I should have said precision.
If I reformulate my original question, I would ask:
Are the 16-bit values from the CMRR sequence are true values from the ADC, i.e. not obtained from scaling the interval [0,4095] to [0,65535]? Or are they values obtained by scaling like if I used dcm2niix with option -l (lossless scaling)?
My guess is that since the ADC is capable of 32-bit sampling, it is likely that the 16-bit values are true values from the reconstruction. But I might be wrong.

**Reply by mharms** [reply]
@eauerbach will have to confirm, but I'm >95% confident that the 16-bit DICOMs reflect "true values" during the reconstruction, rather than just a scaling of 12-bit values back to 16-bit.  There would be little point in doing the latter.

**Reply by eauerbach** [reply]
For most of the image reconstruction, the data are held as 32-bit floating point. Only at the very end of the reconstruction is it scaled and converted to 16-bit short integer. During the conversion to 16-bit it can optionally be scaled more aggressively and truncated to 12-bit for compatibility with old software (including some of the Siemens post processing tools!).

Clicking the "suppress 16-bit DICOM" box only toggles whether the last step is performed. There is no reason to select it unless you have to use some old software that is not 16-bit compatible. It doesn't even save space, because the 12-bit data are still stored in 16-bit fields.

Legend holds the other 4 bits used to be reserved for overlays and masks, but I have never encountered that. 12 bits was probably just inherited from CT since it is enough for displaying clinical images in greyscale where you have predictable dynamic range, but it isn't great for quantitative MR imaging.

**Reply by jberrebi** [reply]
Thank you @eauerbach  for the detailed answer! Then I know that checking the box is not worth it for estimating "real-time" motion. I'll have to find another way.

**Comment by jmatthews-rotman** [comment]
Apologies if I missed this somewhere else, but are there any update notes for pre10 and pre11?

**Reply by eauerbach** [reply]
Sorry I have not been updating this topic reliably, but I have updated the [Release Notes](https://www.cmrr.umn.edu/multiband/Multi-Band_C2P_Instructions_R017.pdf) with this information.

**Comment by Matej-Kudrna** [comment]
Dear All, last week we upgraded Skyra to XA60, I tried to install the sequence, but wasn't able to proceed due to this window: ![PXL_20250416_043858210.jpg](https://github.com/user-attachments/assets/92280b7c-e837-4667-9452-21801457212d)

I installed the eja certificate, restarted the system, before trying to install the sequence. Our Siemens technician told me that this is not their problem. I even excluded the installer from Firewall.

Any suggestions?

Thanks Matej

**Comment by xiangruili** [comment]
Do you get the same error message if you run the installed from a Command Window with Administrator privilege?

**Comment by sjayejohnston** [comment]
Thanks Ross, that's great.

Best

Steve

**Comment by tongluobing** [comment]
Hi,

I noticed that the X20A is missing in the latest release. Could you please let me know when a version for the X20A will be available again?

Thank you very much.
Jian

**Reply by jmatthews-rotman** [reply]
I see "Release R017pre13 for XA20A" listed on the download page here:
https://downloads.cmrr.umn.edu/software/package/multiband/

**Reply by tongluobing** [reply]
Thanks a lot for sending me the correct link – I’d been trying to find the release from the wrong one.

Best,
Jian

**Comment by jZhuoMRI** [comment]
I just installed R017pre13 on our VE11E Prisma scanner and noticed that I cannot find the option to turn on “EPI noise scans” under the Sequence → Special card.

I was shared a protocol file where “EPI noise scans” is set to 3, so I’m wondering if this is due to a sequence version difference, or if this option is not available on the VE11E platform.

Thank you very much.

Chenny

**Comment by BPaton** [comment]
There is an option in the ini file for turning this on. Does the ini file
still exist in the right location eg the customer sequence directory?

https://github.com/CMRR-C2P/MB/wiki

Bryan.

**Reply by jZhuoMRI** [reply]
Thank you very much. It worked. 

Chenny

**Comment by jZhuoMRI** [comment]
I’m trying to build a multi-echo fMRI protocol based on a PDF shared by another site running XA30. We are currently on VE11E with a Prisma. I’ve matched essentially all parameters, including: BW (2470 Hz/Px), echo spacing (0.51 ms), EPI factor (88), FoV (220 mm, 100%), base resolution (88), phase resolution (100%), phase partial Fourier (6/8), reference lines PE (16), acceleration (GRAPPA factor 2), reference scan mode (FLEET), and excitation pulse duration (5270 µs).

However, I can’t figure out why our minimum TE is 17.20 ms (17.20, 35.14, 53.06, 71.00, 88.92), while theirs is 14.20 ms (14.20, 31.62, 49.04, 66.46, 83.88).

Given that echo spacing and excitation pulse duration are the same, I’m wondering what else could account for this ~3 ms difference in minimum TE. Are there any parameters I might be missing or overlooking?

Any suggestions would be greatly appreciated. Thanks very much.

Chenny

**Reply by jmatthews-rotman** [reply]
Is your Gradient Mode matched? If they're on Performance and you're on Fast it might slow you down, although I would have expected that to limit your Echo Spacing as well. Check the Exite Pulse Duration and other sequence specific settings in the Special Card as well. If you have contact with the site, you could also just ask if they can export an exar for you to import. You mentioned the other site is on XA30 but not what scanner they have. If they're one of the rare Cima sites, they're increased gradient performance might just be matchable on a Prisma. I think most of the Cima's are probably on XA60 though.

**Comment by rwmair** [comment]
Eddie,
Any updates to the timeline for XB10 sequences?
I have collaborators who are doing a Vida-fit upgrade next to month to their old Skyra, the Vida console is reportedly shipping with XB10.
Thanks!

**Reply by eauerbach** [reply]
It is available now.
